pkg/v1/config/clientconfig.go

```diff
@@ -37,6 +37,11 @@ const (
 	// when creating a cluster.
 	FeatureFlagManagementClusterCustomNameservers = "features.management-cluster.custom-nameservers"
 	FeatureFlagClusterCustomNameservers           = "features.cluster.custom-nameservers"
+	// Network Separation feature flags determine whether it is permitted to
+	// provide the AVI_MANAGEMENT_CLUSTER_SERVICE_ENGINE_GROUP, AVI_CONTROL_PLANE_NETWORK, AVI_CONTROL_PLANE_NETWORK_CIDR,
+	// AVI_MANAGEMENT_CLUSTER_CONTROL_PLANE_VIP_NETWORK_NAME and AVI_MANAGEMENT_CLUSTER_CONTROL_PLANE_VIP_NETWORK_CIDR
+	// when creating a cluster.
+	FeatureFlagManagementClusterNetworkSeparation = "features.management-cluster.network-separation"
```

Comment:
Should we add this flag with a beta flag notation? cc @swalner-vmware 

@iXinqi are we expecting to enable this feature to `true` by default sometime in the future when we have the support?

Reply:
@anujc25 yes, this feature will be true by default in the future, after it's been fully tested.
What is beta flag notation?

Reply:
@iXinqi , the idea of a "beta" flag is to avoid this scenario:
A user installs the CLI with your flag set to false (by default); let's use the example of `feature.cluster.foobar`. They never change the value. Later when they install a newer release where `feature.cluster.foobar` is true by default, we have no way of overriding the false setting in their config file, because it appears identical to the case where they intentionally set the file false (and we should NOT override the value in that case). (We have no installation script that would help avoid this problem, so we'd have to tell the user to run a specific command to override their previous value.)
To avoid this issue we recommend having two flags. The "beta" flag, e.g. `feature.cluster.foobar-beta` is installed during the beta period with a value of false. Later, when the feature is ready to be released with a default of value of true, the developer changes the definition of the flag to `feature.cluster.foobar`. This is a "new" flag in terms of the config file, so when the new software runs, the old `feature.cluster.foobar-beta` flag in the config file is totally ignored, and now we can set the default of `feature.cluster.foobar` to true (when we don't find it in the config file).

I hope the above is clear, but I'm happy to chat directly about it (if that would help).

Reply:
@swalner-vmware Do you think we must add the beta flag right now? Since Tanzu RC4 is due sometime this week, is it possible to merge this pr and then we'll submit a new pr for the bata flag after Eretria RTM?

Reply:
My suggestion is simply to change (in this PR):
`FeatureFlagManagementClusterNetworkSeparation = "features.management-cluster.network-separation"`
to 
`FeatureFlagManagementClusterNetworkSeparation = "features.management-cluster.network-separation-beta"`

Reply:
If you have an installation script that touches this flag, or documentation of it, then those things would have to change as well. Otherwise, you should be fine.

Reply:
Thanks @swalner-vmware, I updated the flag. May I have your review again?